src/util/symbol.cpp

```diff
@@ -112,3 +113,25 @@ symbol_exprt symbolt::symbol_expr() const
 {
   return symbol_exprt(name, type);
 }
+
+/// Check that the instance object is well formed.
+/// \return: true if well-formed; false otherwise.
+bool symbolt::is_well_formed() const
```

Comment:
I think we're calling this 'validate' in most other places.

Reply:
I'd expect `validate` to throw, whereas `is_valid` would return a boolean FWIW

Reply:
I vastly prefer this name, and makes more logical sense. `validate` is a verb, which implies that the `symbol` object is applying an action on something else. `is_something` for any kind of value of something is well established convention for predicates like this.